torch/csrc/lazy/ts_backend/dynamic_ir.cpp

```diff
@@ -36,6 +36,13 @@ SizeNode::SizeNode(Value input, size_t dim):
 int64_t SizeNode:: getStaticValue() const {
     return dynamic_cast<const TsNode*>(operand(0).node)->shape(0).size(dim_);
 }
+bool SizeNode:: isDynamic() const {
+    auto symbolic_vec = dynamic_cast<const TsNode*>(operand(0).node)->shape(0).is_symbolic();
+    if(!symbolic_vec.has_value()){
```

Comment:
what's the reasoning behind this logic?

Reply:
The idea is that if the `symbolic_vec` is `nullopt`, it either is not calculatable, or it is an item with dynamic dims. The conservative thing to do in that case would be to mark it as symbolic.

Reply:
Not calculatable is either missing JIT functionality, and for the rare case where there are dynamic dims.